scrapper.py

```diff
@@ -23,75 +36,310 @@ class IncorrectNumberOfArticlesError(Exception):
+        }
+
+        month, day, year = date_str.split()
+        day = int(day[:-1])
+        year = (int(year))
+        month = month_dict[month]
+        return datetime(year, month, day)
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        article_page = Crawler.get_page(self.article.url)
+        soup = BeautifulSoup(article_page, 'html.parser')
+
+        # with open(os.path.join(constants.ASSETS_PATH, f'{self.article.article_id}_page.html'),
+        #           'w', encoding='utf-8') as file:
+        #     file.write(article_page)
 
+        self._fill_article_with_meta_information(soup)
+        self._fill_article_with_text(soup)
+        return self.article
 
-def prepare_environment(base_path):
+
+def prepare_environment():
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if not os.path.exists(constants.PAGES_ROOT):
+        os.makedirs(constants.PAGES_ROOT)
+
+    if not os.path.exists(constants.ASSETS_PATH):
+        os.makedirs(constants.ASSETS_PATH)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    is_config_a_dict = isinstance(config, dict)
+
+    has_config_attributes = (
+        'base_urls' in config and
+        'total_articles_to_find_and_parse' in config
+    )
+
+    if not (is_config_a_dict and has_config_attributes):
+        raise UnknownConfigError
+
+    if 'max_number_articles_to_get_from_one_seed' not in config:
+        config['max_number_articles_to_get_from_one_seed'] = \
+            config['total_articles_to_find_and_parse']
+
+    is_base_urls_correct = (
+            isinstance(config['base_urls'], list) and
+            all([isinstance(seed_url, str) for seed_url in config['base_urls']])
+    )
+
+    if isinstance(config['total_articles_to_find_and_parse'], int):
+        is_total_number_of_articles_correct = config['total_articles_to_find_and_parse'] > 0
+    else:
+        raise IncorrectNumberOfArticlesError
+
+    is_max_number_of_articles_int = (
+        isinstance(config['max_number_articles_to_get_from_one_seed'], int) and
+        config['max_number_articles_to_get_from_one_seed'] > 0
+    )
+
+    is_max_number_of_articles_correct = (
+            config['max_number_articles_to_get_from_one_seed'] <= 10000
+    )
+
+    if not is_base_urls_correct:
+        raise IncorrectURLError
+
+    if not is_total_number_of_articles_correct:
+        raise IncorrectNumberOfArticlesError
+
+    if not is_max_number_of_articles_correct:
+        raise NumberOfArticlesOutOfRangeError
+
+    if all((
+            is_config_a_dict,
+            has_config_attributes,
+            is_base_urls_correct,
+            is_total_number_of_articles_correct,
+            is_max_number_of_articles_int,
+            is_max_number_of_articles_correct
+    )):
+        return config.values()
```

Comment:
are you sure about the order that it is going to be the same across runs? what guarantees that?

Reply:
made it explicit